thetis/tracer_eq_2d.py

```diff
@@ -262,18 +271,33 @@ class TracerEquation2D(Equation):
     """
     2D tracer advection-diffusion equation :eq:`tracer_eq` in conservative form
     """
-    def __init__(self, function_space, depth,
-                 use_lax_friedrichs=False,
-                 sipg_factor=Constant(1.0)):
+    def __init__(self, function_space, depth, options, velocity):
```

Comment:
Ah, I see, you need to have access to the velocity field here. I guess we are now passing it twice, in the eq constructor (here) and then again in the time integrator (via fields). Potential pitfall is that user might define these inconsistently. But I do not see immediately see a better solution.

Reply:
Good point. So long as the user doesn't tamper with the equation and timestepper then I guess they should stay the same, though.

Reply:
Yeah I agree, this doesn't feel quite right - but I can't see a proper solution that isn't much more intrusive. This would be much more of an issue if we would do SUPG for momentum advection....